File: src/php/pe-007.php
<?php

function Prime($num) {
	$high = round(sqrt($num));
	if ($high >= 2) {
		foreach (range(2, $high) as $i) {
			if ($num%$i == 0) {
				return false;
			}
		}
	}
	return true;
}

function FindPrime($ending) {
	$primes = array(2);
	$i = 3;
	while (true) {
		if (Prime($i)) {
			array_push($primes, $i);
			if (count($primes) == $ending) {
				echo end($primes);
				break;
			}
		}
		$i+=2;
	}
}

FindPrime(10001);

?>

File: src/py/pe-007.py
import math

def Prime(num):
	high = round(math.sqrt(num))
	for i in range(2, num):
		if (num%i==0):
			return False

	#else return true
	return True

#Find Prime
def FindPrime(ending):
	primes = [2]
	i = 3
	while True:
		if (Prime(i)):
			primes.append(i)
			if len(primes) == ending:
				print(primes.pop())
				break
		#Increment
		i += 2

#Process
FindPrime(10001)